lib/chatkit/client.rb

```diff
@@ -769,11 +782,44 @@ def update_permissions_for_role(options)
       })
     end
 
+    def upload_attachment(token, room_id, part)
+      body = part[:file]
+      content_length = body.length
+
+      if content_length <= 0
+        raise Chatkit::MissingParameterError.new("File contents size must be greater than 0")
+      end
+
+      attachment_req = { content_type: part[:type],
+                         content_length: content_length
+                       }
+
+      attachment_response = api_request({
+        method: "POST",
+        path: "/rooms/#{CGI::escape room_id}/attachments",
+        body: attachment_req,
+        jwt: token
+      })
+
+      url = attachment_response[:body][:upload_url]
+      connection = Excon.new(url, :omit_default_port => true)
+      upload_response = connection.put
+
+      if upload_response.status != 200
+        error = {message: "Failed to upload attachment",
+                 response_object: upload_response
+                }
+        raise Chatkit::UploadError.new(error)
+      end
+
+      attachment_response[:body][:attachment_id]
+    end
+
     def verify(rules, options)
       rules.keys.each { |field_name|
         rule = rules[field_name]
         if options.has_key?(field_name)
-          if not (options[field_name].instance_of? rule[:type])
+          if not (rule[:type].nil? or options[field_name].instance_of? rule[:type])
```

Comment:
Is there a particular reason why you chose to use `or` instead of `||`? Note that `or` has a lower precedence than `||`. 
Another small thing which is stylistic: `options[:field_name].instance_of?(rule[:type])`. The rest of the method calls use `()` to enclose arguments. While ruby allows it, we should be consistent :)

Reply:
Indeed, we do actually have a Ruby style guide (!!), and it says to use parens unless there are no args.

Reply:
https://manual.pusherplatform.io/languages/ruby/style-guide.html

It also bans `and` and `or`, because they are semantically different. In addition to having a lower binding presedence than you might expect, they also don't do short-circuit evaluation, but always fully evaluate both sides of the expression.

Reply:
Welcome to Ruby ;)

Reply:
:+1: